### PR TITLE
Introduce per file dax support and no_readdir config option

### PR DESCRIPTION
**Pull request by eryugey**
- api: impl Default for Entry structure
- passthroughfs: support per-file DAX
- abi: add with_flags function to create linux_abi::Attr
- passthrough: don't flood log with unnecessary error logs
- passthrough: introduce no_readdir config

### CODE REVIEW

File: src/passthrough/async_io.rs [hunk 1]
```diff
@@ -314,6 +314,17 @@ impl<D: AsyncDrive + Sync, S: BitmapSlice + Send + Sync> AsyncFileSystem<D, S>
             })
             .await?;
 
+        let mut attr_flags: u32 = 0;
+        if let Some(dax_file_size) = self.cfg.dax_file_size {
+            // st.stat.st_size is i64
+            if self.perfile_dax.load(Ordering::Relaxed)
+                && st.stat.st_size >= 0x0
```
Comment (jiangliu) [inline src/passthrough/async_io.rs:321]: is it possible that `st_size` has negative value?

File: src/passthrough/async_io.rs [hunk 2]
```diff
@@ -314,6 +314,17 @@ impl<D: AsyncDrive + Sync, S: BitmapSlice + Send + Sync> AsyncFileSystem<D, S>
             })
             .await?;
 
+        let mut attr_flags: u32 = 0;
+        if let Some(dax_file_size) = self.cfg.dax_file_size {
+            // st.stat.st_size is i64
+            if self.perfile_dax.load(Ordering::Relaxed)
```
Comment (jiangliu) [inline src/passthrough/async_io.rs:320]: Why uses two flags "self.cfg.dax_file_size" and "self.perfile_dax" to control the perfile-dax?

Comment (jiangliu) [reply src/passthrough/async_io.rs:320]: Aha, got it. The `perfile_dax` reflects host/guest negotiating result.

**Review by jiangliu — APPROVED** [review]

File: src/passthrough/async_io.rs (hunk 1 above)
Comment (eryugey) [reply src/passthrough/async_io.rs:321]: It seems not possible from stat64(2), but I think it's no harm to be defensive.

**Pull request merged**